flocker/provision/_gce.py

```diff
@@ -272,32 +297,71 @@ def _create_gce_instance_config(instance_name, project, zone, machine_type,
     return gce_slave_instance_config
 
 
-@implementer(INode)
-class GCENode(PClass):
+class GCEInstance(PClass):
     """
-    ``INode`` implementation for GCE.
+    Simple GCE instance object that represents an instance as a cloud resource.
+
+    This does not contain any information about the distribution running on the
+    instance or credentials that can be used to connect to the node.
 
     :ivar unicode address: The public IP address of the instance.
     :ivar unicode private_address: The network internal IP address of the
         instance.
-    :ivar bytes distribution: The OS distribution of the instance.
     :ivar unicode project: The project the instance a member of.
     :ivar unicode zone: The zone the instance is in.
     :ivar unicode name: The GCE name of the instance used to identify the
         instance.
-    :ivar bytes username: The preferred username to access the instance as.
     :ivar compute: A Google Compute Engine Service that can be used to make
         calls to the GCE API.
     """
     address = field(type=bytes)
     private_address = field(type=bytes)
-    distribution = field(type=bytes)
     project = field(type=unicode)
     zone = field(type=unicode)
     name = field(type=unicode)
-    username = field(type=bytes)
     compute = field()
 
+    def destroy(self):
+        """
+        Destroy the instance. Blocks until the operation is completed.
+        """
+        with start_action(
+            action_type=u"flocker:provision:gce:destroy",
+            instance_id=self.name,
+        ):
+            operation = self.compute.instances().delete(
+                project=self.project,
+                zone=self.zone,
+                instance=self.name
+            ).execute()
+            wait_for_operation(self.compute, operation, timeout_steps=[5]*60)
```

Comment:
Nice catch, you're keeping me honest :).

Yea I did see it take somewhere between 3 and 4 minutes once when I ran the test.

As this is test infra code, the right answer seemed to be just wait up to 5 minutes. I also figured if it was going to take that long we probably don't need to poll every second.

Added a comment so people don't think about making it shorter.